src/firefly/js/visualize/saga/CoverageWatcher.js

```diff
@@ -334,8 +334,8 @@ function updateCoverage(tbl_id, viewerId, preparedTables, options, tblCatIdMap,
                     if (allRowsTable?.tableData?.data?.length > 0) {
                         preparedTables[tbl_id] = allRowsTable;
                         const isRegion = isTableWithRegion(allRowsTable);
-
-                        tblCatIdMap[tbl_id] = (isRegion) ? [centerId(tbl_id), regionId(tbl_id)] : [tbl_id, searchTargetId(tbl_id)];
+                        const regionAry = getRegionAryFromTable(options, table);
```

Comment:
I am concerned this this fairly expensive call is being call to often. It should be wrapped with `memorizeLastCall` (from WebUtil.js)